src/Makefile.am

```diff
@@ -9,13 +9,13 @@ lib_LTLIBRARIES = libica.la
 libica_la_CFLAGS = ${AM_CFLAGS} -I${srcdir}/include -I${srcdir}/../include \
 		   -fvisibility=hidden -pthread
 libica_la_CCASFLAGS = ${AM_CFLAGS}
-libica_la_LIBADD = @LIBS@ -lrt -lcrypto
+libica_la_LIBADD = @LIBS@ -lrt -lcrypto -ldl
 libica_la_LDFLAGS = -Wl,--version-script=${srcdir}/../libica.map \
 		    -version-number ${VERSION}
 libica_la_SOURCES = ica_api.c init.c icastats_shared.c s390_rsa.c \
 		    s390_crypto.c s390_ecc.c s390_prng.c s390_sha.c \
 		    s390_drbg.c s390_drbg_sha512.c test_vec.c fips.c \
-		    mp.S rng.c \
+		    mp.S rng.c fips_check.c \
```

Comment:
we already have a fips.c, any reason why the fips intergrity check needs a seperate source file ?

Reply:
Yes, because this integrity check comprises 5 routines and a couple of definitions, I put it into a separate file. But there is no other reason why not adding the code to fips.c

Reply:
Id put everything in fips.c/.h

Reply:
Done.